src/lib.rs

```diff
@@ -287,3 +287,15 @@ pub use self::decode_all::DecodeAll;
 pub use self::depth_limit::DecodeLimit;
 pub use self::encode_append::EncodeAppend;
 pub use self::encode_like::{EncodeLike, Ref};
+
+/// Assert Decode::decode and Decode::skip works
+///
+/// NOTE: this is not part of public API. Only used for internal tests
+#[doc(hidden)]
+pub fn assert_decode<T>(mut encoded: &[u8], res: T) where
```

Comment:
Why not put this in the `tests/` folder (or in the utils)?

Reply:
> Why not put this in the `tests/ `

Because I also use in tests in `src/*`

> (or in the utils)?

yeah I guess we can move to its own module but I think it is fine here as well